pkg/test/e2e/expose-strategy/client.go

```diff
@@ -0,0 +1,167 @@
+/*
+Copyright 2021 The Kubermatic Kubernetes Platform contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package exposestrategy
+
+import (
+	"encoding/json"
+	"fmt"
+	"strconv"
+	"strings"
+
+	e2eutils "k8c.io/kubermatic/v2/pkg/test/e2e/utils"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/pointer"
+
+	"k8c.io/kubermatic/v2/pkg/resources"
+	"k8c.io/kubermatic/v2/pkg/semver"
+)
+
+const (
+	openvpnImage        = "quay.io/kubermatic/openvpn:v2.5.0-r1"
+	clientPodName       = "client-pod"
+	clientContainerName = "openvpn"
+	kubeconfigPath      = "/etc/kubernetes"
+)
+
+// KubeVersions is used to unmarshal the output of 'kubectl version -ojson'
+type KubeVersions struct {
+	ServerVersion KubeVersion `json:"serverVersion"`
+	ClientVersion KubeVersion `json:"clientVersion"`
+}
+
+type KubeVersion struct {
+	Major      string                 `json:"major"`
+	Minor      string                 `json:"minor"`
+	GitVersion string                 `json:"gitVersion"`
+	Other      map[string]interface{} `json:"-"`
+}
+
+func (k KubeVersion) MajorVersion() uint64 {
+	if v, err := strconv.ParseInt(k.Major, 10, 64); err == nil {
+		return uint64(v)
+	}
+	return 0
+}
+
+func (k KubeVersion) MinorVersion() uint64 {
+	if v, err := strconv.ParseInt(k.Minor, 10, 64); err == nil {
```

Comment:
The major and minor version strconv methods can be a single function used in both parts

Reply:
Do you mean something like:
```
func (k KubeVersion) MinorVersion() (major uint64, minor uint64)
```
?